tests/run-macros/i6988/FirstArg_1.scala

```diff
@@ -0,0 +1,33 @@
+package foo
+
+case class FirstArg(value: Any, source: String)
+object FirstArg {
+  inline given as FirstArg = ${Macros.argsImpl}
+}
+
+object Macros {
+  import language.experimental.macros
+  import scala.quoted._
+  import scala.tasty._
+
+  def argsImpl given (qctx: QuoteContext): Expr[FirstArg] = {
+    import qctx.tasty._
+    var enclosingCls: ClassDefSymbol = rootContext.owner.asInstanceOf[ClassDefSymbol]
+    def enclosingParamList(owner: Symbol): Seq[Seq[Symbol]] =
+      owner match {
+        case IsClassDefSymbol(x) =>
+          enclosingCls = x
+          x.tree.constructor.paramss map { _ map {
+            _.symbol
+          }}
+        case _ =>
+          enclosingParamList(owner.owner)
+      }
+    def literal(value: String): Expr[String] =
+      Literal(Constant(value)).seal.asInstanceOf[Expr[String]]
+    val paramss = enclosingParamList(rootContext.owner)
+    val firstArg = paramss.flatten.head
+    val ref = Select.unique(This(enclosingCls), firstArg.name)
```

Comment:
Not sure if this is a fix, since my report is about `Ref(...)` not working.

Reply:
The issue is not with `Ref`, but about compiler internals: at the code place & pickling phase where we refer to the constructor symbol, it's simply not accessible.

After the compiler phases that actually create primary constructors, the code might work, but it's far after pickling.

Reply:
We should also have an assertion inside `Ref` to prevent the creation of an inconsistent ref.

Reply:
I don't think it's practical to add checks in `Ref` --- it works as expected, it's where we put the tree result in pickling crash.